tests/filecheck/backend/rvscf_lowering_labels.mlir

```diff
@@ -0,0 +1,112 @@
+// RUN: xdsl-opt -p lower-riscv-scf-to-labels --split-input-file %s | filecheck %s
+
+builtin.module {
+    "riscv_func.func"() ({
+    ^0(%0 : !riscv.reg<a0>, %1 : !riscv.reg<a1>):
+        %2 = riscv.mv %0 : (!riscv.reg<a0>) -> !riscv.reg<a0>
+        %3 = riscv.mv %1 : (!riscv.reg<a1>) -> !riscv.reg<a1>
+        %4 = riscv.li 1 : () -> !riscv.reg<a2>
+        %5 = riscv.li 0 : () -> !riscv.reg<a3>
+        %6 = riscv.fcvt.s.w %5 : (!riscv.reg<a3>) -> !riscv.freg<fa0>
+        %7 = "riscv_scf.for"(%2, %3, %4, %6) ({
+        ^1(%8 : !riscv.reg<a0>, %9 : !riscv.freg<fa0>):
+            %10 = riscv.fcvt.s.w %8 : (!riscv.reg<a0>) -> !riscv.freg<fa1>
+            %11 = riscv.fadd.s %9, %10 : (!riscv.freg<fa0>, !riscv.freg<fa1>) -> !riscv.freg<fa0>
+            "riscv_scf.yield"(%11) : (!riscv.freg<fa0>) -> ()
+        }) : (!riscv.reg<a0>, !riscv.reg<a1>, !riscv.reg<a2>, !riscv.freg<fa0>) -> !riscv.freg<fa0>
+        %12 = riscv.fcvt.w.s %7 : (!riscv.freg<fa0>) -> !riscv.reg<a0>
+        "riscv_func.return"(%12) : (!riscv.reg<a0>) -> ()
+    }) {"sym_name" = "foo"} : () -> ()
+}
+
+// CHECK:       builtin.module {
+// CHECK-NEXT:    "riscv_func.func"() ({
+// CHECK-NEXT:    ^0(%0 : !riscv.reg<a0>, %1 : !riscv.reg<a1>):
+// CHECK-NEXT:      %2 = riscv.mv %0 : (!riscv.reg<a0>) -> !riscv.reg<a0>
+// CHECK-NEXT:      %3 = riscv.mv %1 : (!riscv.reg<a1>) -> !riscv.reg<a1>
+// CHECK-NEXT:      %4 = riscv.li 1 : () -> !riscv.reg<a2>
+// CHECK-NEXT:      %5 = riscv.li 0 : () -> !riscv.reg<a3>
+// CHECK-NEXT:      %6 = riscv.fcvt.s.w %5 : (!riscv.reg<a3>) -> !riscv.freg<fa0>
+// CHECK-NEXT:      %7 = riscv.get_register : () -> !riscv.reg<a0>
+// CHECK-NEXT:      riscv.label "scf_cond_0_for" ({
+// CHECK-NEXT:      }) : () -> ()
+// CHECK-NEXT:      riscv.bge %7, %3, "scf_body_end_0_for" : (!riscv.reg<a0>, !riscv.reg<a1>) -> ()
+// CHECK-NEXT:      riscv.label "scf_body_0_for" ({
+// CHECK-NEXT:      }) : () -> ()
+// CHECK-NEXT:      %8 = riscv.get_float_register : () -> !riscv.freg<fa0>
+// CHECK-NEXT:      %9 = riscv.get_register : () -> !riscv.reg<a0>
+// CHECK-NEXT:      %10 = riscv.fcvt.s.w %9 : (!riscv.reg<a0>) -> !riscv.freg<fa1>
+// CHECK-NEXT:      %11 = riscv.fadd.s %8, %10 : (!riscv.freg<fa0>, !riscv.freg<fa1>) -> !riscv.freg<fa0>
+// CHECK-NEXT:      %12 = riscv.add %7, %4 : (!riscv.reg<a0>, !riscv.reg<a2>) -> !riscv.reg<a0>
+// CHECK-NEXT:      riscv.blt %7, %3, "scf_body_0_for" : (!riscv.reg<a0>, !riscv.reg<a1>) -> ()
+// CHECK-NEXT:      riscv.label "scf_body_end_0_for" ({
+// CHECK-NEXT:      }) : () -> ()
+// CHECK-NEXT:      %13 = riscv.get_float_register : () -> !riscv.freg<fa0>
+// CHECK-NEXT:      %14 = riscv.fcvt.w.s %13 : (!riscv.freg<fa0>) -> !riscv.reg<a0>
+// CHECK-NEXT:      "riscv_func.return"(%14) : (!riscv.reg<a0>) -> ()
+// CHECK-NEXT:    }) {"sym_name" = "foo"} : () -> ()
+// CHECK-NEXT:  }
+
+// -----
```

Comment:
Is there a reason we are using --split-input-file? AFAIU this makes it harder to run individual test cases with `xdsl-opt`, no?

Reply:
It's a tradeoff, using `--split-input-file` lets us reuse the loaded instance of the xdsl module, which takes hundreds of ms to import, which adds up, but more importantly it grous the tests by semantical meaning, so I find it a bit easier to navigate. You're right that it has its drawbacks also.